src/libraries/Common/src/Internal/Cryptography/AsymmetricAlgorithmHelpers.Der.cs

```diff
@@ -69,18 +192,75 @@ private static void CopySignatureField(ReadOnlySpan<byte> signatureField, Span<b
         {
             if (signatureField.Length > response.Length)
             {
-                // The only way this should be true is if the value required a zero-byte-pad.
-                Debug.Assert(signatureField.Length == response.Length + 1, "signatureField.Length == fieldLength + 1");
-                Debug.Assert(signatureField[0] == 0, "signatureField[0] == 0");
-                Debug.Assert(signatureField[1] > 0x7F, "signatureField[1] > 0x7F");
+                if (signatureField.Length != response.Length + 1 ||
+                    signatureField[0] != 0 ||
+                    signatureField[1] <= 0x7F)
```

Comment:
Response is always field-size wide; so an empty response would mean a zero-bit key.

SignatureField can't represent zero, for algorithm-conforming types, because neither r nor s are allowed to be zero in either DSA or ECDSA.